grid-client/deployer/network_deployer.go

```diff
@@ -318,36 +432,29 @@ func (d *NetworkDeployer) Deploy(ctx context.Context, znet *workloads.ZNet) erro
 
 // BatchDeploy deploys multiple network deployments using the deployer
 func (d *NetworkDeployer) BatchDeploy(ctx context.Context, znets []*workloads.ZNet, updateMetadata ...bool) error {
-	newDeployments := make(map[uint32][]gridtypes.Deployment)
-	newDeploymentsSolutionProvider := make(map[uint32][]*uint64)
-	nodePorts := make(map[uint32][]uint16)
+	filteredZNets := make([]*workloads.ZNet, 0)
 	var multiErr error
+	if err := validateAccountBalanceForExtrinsics(d.tfPluginClient.SubstrateConn, d.tfPluginClient.Identity); err != nil {
+		return err
+	}
 	for _, znet := range znets {
-		err := d.Validate(ctx, znet)
-		if err != nil {
-			multiErr = multierror.Append(multiErr, fmt.Errorf("invalid network %s: %w", znet.Name, err))
+		if err := znet.Validate(); err != nil {
+			multiErr = multierror.Append(multiErr, err)
 			continue
 		}
-
-		dls, err := d.GenerateVersionlessDeployments(ctx, znet, nodePorts)
-		if err != nil {
-			multiErr = multierror.Append(multiErr, fmt.Errorf("could not generate deployments data for network %s: %w", znet.Name, err))
+		if err := d.InvalidateBrokenAttributes(znet); err != nil {
```

Comment:
I think `InvalidateBrokenAttributes` and `validateAccountBalanceForExtrinsics` are included in `Validate` function

Reply:
yep but `Validate` will does some extra requests to nodes that we already do as part of network processing, it also will get balance multiple times so it's way slower since it is done for each network

Reply:
yes that's why I am suggesting refactoring it. to validate balance once then validate all znets you pass to it